pkg/common/common.go

```diff
@@ -239,3 +240,53 @@ func GetImagePullPolicy(image string) (corev1.PullPolicy, error) {
 	}
 	return corev1.PullAlways, nil
 }
+
+// GenerateCliArgsFromConfigMap generates CLI arguments from a ConfigMap.
+//
+// This function takes a CLI subcommand and a ConfigMap, and returns a slice of strings representing
+// the CLI arguments from the ConfigMap.
+// The function processes each key-value pair in the ConfigMap as follows:
+//
+//  1. If the ConfigMaps' value starts and ends with single (') or double (") quotes, it is left unchanged.
+//  2. If the ConfigMaps' key starts with single '-' or double '--', it is left unchanged.
+//  3. If an argument key is passed without `-` or `--` prefix, then `--` is added as a key prefix.
+//  4. If the ConfigMap value is "true" or "false" (case-insensitive), it is converted to lowercase
+//     and used without single quotes surroundings (boolean value).
+//  5. The formatted key-value pair is added to the result.
+//
+// Args:
+//
+//	cliSubCommand: The CLI subcommand as a string, for example 'server'
+//	configMap: A pointer to a corev1.ConfigMap containing key-value pairs.
+//
+// Returns:
+//
+//	A slice of strings representing the CLI arguments.
+func GenerateCliArgsFromConfigMap(cliSubCommand string, configMap *corev1.ConfigMap) []string {
```

Comment:
If it's surrounded by only one " or ' then it's ending up as string, so for ConfigMap:

```shell
# ConfigMap:
# myValue: 'myVal

# we will have broken arg:

$ velero server myValue=''myVal'
```

The above will break CLI argument, however for the:

```shell
# ConfigMap:
# myValue: "myVal

# we will have a valid arg:

$ velero server myValue='"myVal'
```

I am not sure if we want to mess parsing strings so much... ?

Reply:
I am thinking, lets strip characters like `", ', `` from the values and add the quotes ourselves, WDYT ? @mpryc

Reply:
I think we should not, because we will end up having a lot of validations to make the value `shell` compatible. There are many other gotchas to consider and we would have to take them all what would create possible bugs.

I really do not want to manipulate what user entered as it's unsupported arg and it should go as is.

Here is just a short list of characters to be escaped that comes to my mind and there may be more, so really dunno if we want to go there...

```
'
"
\
$
`
(, )
*
?
[, ]
|
&
;
<
>
\n
space
```

Reply:
I understand your point but how likely is the user to make mistake regarding other characters in comparison with single quotes/double quotes ? The probability is the highest for the quotes characters as compared to others. Hence the thought.

Reply:
Right, but this is unsupported args which is already very custom, so the user will need to be double careful of what args are there, I believe the risk either way is very small. Removing single quite from the beginning if there is no ending and double quote is something we can do, so your call :)

Reply:
When I think more about this, what if someone enters:
```
ConfigMap:
  myarg: 'my value' value
```

This would be converted by us to:
```
$ cli server --myarg='my value' value'
```

Remember we removed first char as it didn't have ending ' and wrapped around single quotes to make sure it's string. Really I think we should not do any magic here with edits of what user inputs

Reply:
@shubham-pampattiwar updated function, tested with your ConfigMap as well the ConfigMap that consists of malformed data and everything is passed as string so there is no risk of breakage:

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: oadp-unsupported-velero-server-args
  namespace: openshift-adp
data:
  default-volume-snapshot-locations: $(echo cat /proc/1/cmdline)
  features: EnableCSI
  uploader-type: kopia

```